kv/memberlist/memberlist_client_test.go

```diff
@@ -989,14 +1007,15 @@ func runClient(kv *Client, name string, ringKey string, portToConnect int, casIn
 	// stop gossipping about the ring(s)
 	defer services.StopAndAwaitTerminated(context.Background(), kv.kv) //nolint:errcheck
 
+	localhostBindAddrsOnce.Do(setLocalhostBindAddrs)
 	for {
 		select {
 		case <-start:
 			start = nil
 
 			// let's join the first member
 			if portToConnect > 0 {
-				_, err := kv.kv.JoinMembers([]string{fmt.Sprintf("127.0.0.1:%d", portToConnect)})
+				_, err := kv.kv.JoinMembers([]string{localhostIP.String() + fmt.Sprintf(":%d", portToConnect)})
```

Comment:
Likely going to be a conflict between this and the "join host port" PR, I will resolve once that is merged.